.studio/chef-server-collection

```diff
@@ -392,6 +392,76 @@ function test_knife() {
   done
 }
 
+
+document "test_ocid" <<DOC
+  test OCID against the embedded Chef server
+DOC
+function test_ocid() {
+  check_if_deployinate_started || return 1
+  bootstrap_chef_user_data || return 1
+
+  test_if_webui_key_is_patched
+  test_if_env_vars_are_configured
+}
+
+test_if_env_vars_are_configured() {
+  ocid_config_file_path="$(hab pkg path chef/oc_id)/oc_id/config/settings/production.yml"
+
+  # Read settings/production.yml into an associative array
+  declare -A config_values
+  while IFS=":" read -r key value; do
+    config_values["$key"]="${value// /}"  # Remove spaces from the value
+  done < "$ocid_config_file_path"
+
+  errors=()
+
+  if [[ "${config_values[endpoint]}" != "https://127.0.0.1:443" ]]; then
+    errors+=("Endpoint not configured properly.")
+  fi
+
+  if [[ "${config_values[superuser]}" != "pivotal" ]]; then
+    errors+=("Superuser not configured properly.")
+  fi
+
+  if [[ "${config_values[ssl_verify_mode]}" != "verify_none" ]]; then
+    errors+=("Verify mode not configured properly.")
+  fi
+
+  if [[ ${#errors[@]} -gt 0 ]]; then
+    printf '%s\n' "${errors[@]}"
+    return 1
+  fi
+}
+
+test_if_webui_key_is_patched() {
+  install_if_missing core/jq-static jq
+
+  # Extract webui_key from private-chef-secrets.json
+  ocid_secrets_file_path="$(hab pkg path chef/oc_id)/oc_id/config/private-chef-secrets.json"
+  ocid_webui_key="$(jq --arg variable_name "chef-server" -r '.[$variable_name]."webui_key"' < $ocid_secrets_file_path)"
+
+  # Create a temporary file with the ocid_webui_key content
+  echo "$ocid_webui_key" > temp_ocid_webui_key.pem
+
+  # Path for erchef webui_priv.pem file
+  erchef_webui_key_path="/hab/svc/automate-cs-oc-erchef/data/webui_priv.pem"
+
+  # Compare the temporary file with the erchef PEM file
+  output=$(diff "$erchef_webui_key_path" <(cat temp_ocid_webui_key.pem))
+
+  # Check if the output is empty, indicating a match
+  if [ -z "$output" ]; then
+    echo "The content of the PEM file matches."
+    return 1
+  else
+    echo "The content of the PEM file does not match."
+    return 0
```

Comment:
exit 0 is success and exit 1 is failure. Pls check the exit code properly.

Reply:
I have removed the return 0 statements from the test blocks. Now we only have return 1 at the time of failures.